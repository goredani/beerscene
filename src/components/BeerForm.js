import React from 'react';
import moment from  'moment';


let regex = /^\d{1,}(\.\d{0,2})?$/;

export default class BeerForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            beername: props.beer ? props.beer.beername : '',
            beertype: props.beer ? props.beer.beertype : '',
            brewery: props.beer ? props.beer.brewery : '',
            description: props.beer ? props.beer.description : '',
            price: props.beer ? (props.beer.price / 100).toString() : '',
            rating: props.beer ? (props.beer.rating / 100).toString() : '',
            ABV: props.beer ? (props.beer.ABV / 100).toString() : '',
            IBU: props.beer ? (props.beer.IBU / 100).toString() : '',
            origin: props.beer ? props.beer.origin : '',
            createdAt: '',
            error: ''
        };
    }

    onBeerNameChange = (e) => {
        const beername = e.target.value;
        this.setState(() => ({beername}));
    };

    onBeerTypeChange = (e) => {
        const beertype = e.target.value;
        this.setState(() => ({beertype}));
    };

    onBreweryChange = (e) => {
        const brewery = e.target.value;
        this.setState(() => ({brewery}));
    };

    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({description}));
    };
    onABVChange = (e) => {
        const ABV = e.target.value;

        if (!ABV || ABV.match(regex)) {
            this.setState(() => ({ ABV }))
        }
    };
    onIBUChange = (e) => {
        const IBU = e.target.value;
        if (!IBU || IBU.match(regex)) {
            this.setState(() => ({IBU}))
        }
    };
    onOriginChange = (e) => {
        const origin = e.target.value;
        this.setState(() => ({ origin }));
    };
 
    onPriceChange = (e) => {
        const price = e.target.value;
        if (!price || price.match(regex)) {
            this.setState(() => ({price}));
        }
    };

    onRatingChange = (e) => {
        const rating = e.target.value;
        if (!rating || rating.match(regex)) {
            this.setState(() => ({rating}));
        }
    };

    onDateChange = (createdAt) => {
        if (createdAt) {
            this.setState(() => ({createdAt}));
        }

    }

    

    onSubmit = (e) => {
        e.preventDefault();

        if (!this.state.beername) {
            this.setState(() => ({error: 'Please provide the name of the beer'}));
        } else if (!this.state.beertype) {
            this.setState(() => ({error: 'Please provide the type of the beer'}));
        } else {
            this.setState(()=> ({ error: ''}));
            this.props.onSubmit({
                beername: this.state.beername,
                beertype: this.state.beertype,
                brewery: this.state.brewery,
                ABV: parseFloat(this.state.ABV, 10 *100),
                IBU: parseFloat(this.state.IBU, 10 *100),
                brewery: this.state.origin,
                description: this.state.description,
                price: parseFloat(this.state.price, 10) *100,
                rating: parseFloat(this.state.rating, 10) *100,
                createdAt: moment().format('YYYY-MM-DD HH:MM')
            });
        }
    };
    
    render() {
        return (
            <div>
                {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.onSubmit}>
                    <input
                        type="text"
                        placeholder="Beer Name"
                        autoFocus
                        value={this.state.beername}
                        onChange={this.onBeerNameChange}
                    />
                    <br />
                    <input
                        type="text"
                        placeholder="Beer Type"
                        value = {this.state.beertype}
                        onChange = {this.onBeerTypeChange}
                    />
                    <br />
                    <input
                        type="text"
                        placeholder="Brewery"
                        value = {this.state.brewery}
                        onChange = {this.onBreweryChange}
                    />
                    <br />
                    <input
                        type="number"
                        placeholder="ABV"
                        value={this.state.ABV}
                        onChange={this.onABVChange}
                    />
                    <br />
                    <input
                        type="number"
                        placeholder="IBU"
                        value={this.state.IBU}
                        onChange={this.onIBUChange}
                    />
                    <br />
                    <input
                        type="text"
                        placeholder="Country Origin"
                        value={this.state.origin}
                        onChange={this.onOriginChange}
                    />
                    <br />
                    <textarea
                        placeholder="Description"
                        value={this.state.description}
                        onChange={this.onDescriptionChange}>
                    </textarea>
                    <br />
                    <input
                        type="number"
                        placeholder="Price"
                        value={this.state.price}
                        onChange={this.onPriceChange}
                    />
                    <br />
                    <input
                        type="number"
                        placeholder="Rating"
                        value={this.state.rating}
                        onChange={this.onRatingChange}
                    />
                    <br />
                    <input
                        type="text"
                        placeholder={moment().format('YYYY, ddd, hA')}
                     />
                  
                    
                   
                    <button>Add Beer</button>
                </form>
            </div>
        )
    }
}

