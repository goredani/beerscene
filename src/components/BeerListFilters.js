import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByABV} from '../actions/filters';


const BeerListFilters = (props) => (
    <div>
        <input 
            type='text' 
            value={props.filters.text} 
            onChange={(e) => {
                props.dispatch(setTextFilter(e.target.value));
            }} 
        />
        <p>Sort By</p>
        <select
            value={props.filters.sortBy}
            onChange={(e) => {
                if (e.target.value === 'date') {
                    props.dispatch(sortByDate());
                } else if (e.target.value === 'ABV') {
                    props.dispatch(sortByABV());
                } 
            }}
        >
            <option value="date">Date</option>
            <option value="ABV">ABV</option>
        </select>
    </div>
);

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};

export default connect(mapStateToProps)(BeerListFilters);