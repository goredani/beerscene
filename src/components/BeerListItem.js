import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import moment from 'moment';

// let recency = (state, props) => {
//   let currentDate = new Date();
//   return(props.beer.createdAt - currentDate);
// };

// console.log(recency());

const BeerListItem = ({ dispatch, id, beername, beertype, brewery, ABV, createdAt, rating, price }) => {
    return(
   <div>
        <h3>Name: {beername}</h3>
        <p>Type: {beertype} - Brewery: {brewery}</p>
        <p>ABV: {ABV}</p>
        <p>Added: {moment(createdAt).format('YYYY-MM-DD')}</p>
        <p>Recency: {moment(createdAt).fromNow()}</p>
        <p>Rating: {rating} - Price: {price} HUF</p>
        
        <Link to={`/edit/${id}`}>
            <button>
                Edit Beer
            </button>
        </Link>
    
    </div>
    
   );
};




export default (BeerListItem);