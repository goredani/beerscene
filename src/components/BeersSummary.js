import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectBeers from '../selectors/beers';
import selectBeersTotal from '../selectors/beers-total';

export const BeersSummary = ({ beersCount, beersTotal }) => {
    const beersWord = beersCount === 1 ? 'beer' : 'beers';
    const formattedBeersTotal = numeral(beersTotal).format('$0,0');
  
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{beersCount}</span> {beersWord} totalling for <span> {formattedBeersTotal} </span> </h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Beer</Link>
                </div>
            </div>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleBeers = selectBeers(state.beers, state.filters);

    return {
        beersCount: visibleBeers.length,
        beersTotal: selectBeersTotal(visibleBeers)
    };
};

export default connect(mapStateToProps)(BeersSummary);