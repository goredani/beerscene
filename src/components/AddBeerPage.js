import React from 'react';
import { connect } from 'react-redux';
import BeerForm from './BeerForm';
import { startAddBeer } from '../actions/beers';

export class AddBeerPage extends React.Component {
  onSubmit = (beer) => {
    this.props.startAddBeer(beer);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <h1>Add Beer</h1>
        <BeerForm
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddBeer: (beer) => dispatch(startAddBeer(beer))
});


export default connect(undefined, mapDispatchToProps)(AddBeerPage);
