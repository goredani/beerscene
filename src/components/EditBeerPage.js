import React from 'react';
import { connect } from 'react-redux';
import BeerForm from './BeerForm';
import { editBeer, startRemoveBeer } from '../actions/beers';


export class EditBeerPage extends React.Component {
  onSubmit = (beer) => {
    this.props.editBeer(this.props.beer.id, beer);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.startRemoveBeer({id: this.props.beer.id});
    this.props.history.push('/');
  }
  render() {
    return (
      <div>
        <BeerForm 
          beer={this.props.beer}
          onSubmit={this.onSubmit}
        />
          <button onClick={this.onRemove} >
          Remove
          </button>
      </div>
    );
   }
};


const mapStateToProps = (state, props) => ({
    beer: state.beers.find((beer) => {
      return beer.id === props.match.params.id;
    })
  
});

const mapDispatchToProps = (dispatch, props) => ({
  editBeer: (id, beer) => dispatch(editBeer(id, beer)),
  startRemoveBeer: (data) => dispatch(startRemoveBeer(data))
});

export default connect(mapStateToProps, mapDispatchToProps)( EditBeerPage);
