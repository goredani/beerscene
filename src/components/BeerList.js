import React from 'react';
import { connect } from 'react-redux';
import BeerListItem from './BeerListItem';
import selectBeers from '../selectors/beers';
import { Link } from 'react-router-dom';

export const BeerList = (props) => (
    <div className="content-container">
        <h1> Live Taplist </h1> 
        <div className="list-header">
            <div>Beer Name</div>
            <div className="show-for-desktop">Beer Type</div>
            <div className="show-for-desktop">Brewery</div>
            <div className="show-for-desktop">IBU</div>
            <div className="show-for-desktop">ABV</div>
            <div>Price</div> 
            <div className="show-for-desktop">Rating</div> 
            <div >Recency</div> 
              
        </div>
        <div className="list-body">
        
        {
            props.beers.length === 0 ? (
                <div className="list-item list-item--message">
                   <div> 
                       <span><p>There are no beers on your taplist.</p></span>
                    </div> 
                    <div>
                      <Link className="button" to="/create">Add Beer</Link>
                    </div>
                </div>
                
            ) : (
                props.beers.map((beer) => {
                    return <BeerListItem key={beer.id} {...beer}/>
                })
            )

        }
        </div>
    </div>
);

const mapStateToProps = (state) => {
    return {
        beers: selectBeers(state.beers, state.filters)
    };
};

export default connect(mapStateToProps)(BeerList);

