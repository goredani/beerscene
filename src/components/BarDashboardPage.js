import React from 'react';
import BeerList from './BeerList';
import BeerListFilters from './BeerListFilters';
import BeersSummary from './BeersSummary';

const BarDashboardPage = () => (
  <div>
    <BeersSummary />
    <BeerListFilters/>
    <BeerList />
  </div>
);

export default BarDashboardPage;
