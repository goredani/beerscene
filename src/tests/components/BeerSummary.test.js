import React from 'react';
import {shallow} from 'enzyme';
import {BeersSummary} from '../../components/BeersSummary';

test('should correctly render BeersSummary with 1 beer', () => {
    const wrapper = shallow(<BeersSummary beerCount={1} beersTotal={195} />);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render BeersSummary with multiple beers', () => {
    const wrapper = shallow(<BeersSummary beerCount={23} beersTotal={195233} />);
    expect(wrapper).toMatchSnapshot();
});

