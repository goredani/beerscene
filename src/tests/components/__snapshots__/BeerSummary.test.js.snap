// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render BeersSummary with 1 beer 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BeersSummary
    beerCount={1}
    beersTotal={195}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Viewing 
         
        beers
         totalling for 
        $195
      </h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Viewing ",
          undefined,
          " ",
          "beers",
          " totalling for ",
          "$195",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Viewing ",
        undefined,
        " ",
        "beers",
        " totalling for ",
        "$195",
      ],
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Viewing 
           
          beers
           totalling for 
          $195
        </h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Viewing ",
            undefined,
            " ",
            "beers",
            " totalling for ",
            "$195",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Viewing ",
          undefined,
          " ",
          "beers",
          " totalling for ",
          "$195",
        ],
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should correctly render BeersSummary with multiple beers 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BeersSummary
    beerCount={23}
    beersTotal={195233}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Viewing 
         
        beers
         totalling for 
        $195,233
      </h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Viewing ",
          undefined,
          " ",
          "beers",
          " totalling for ",
          "$195,233",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Viewing ",
        undefined,
        " ",
        "beers",
        " totalling for ",
        "$195,233",
      ],
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Viewing 
           
          beers
           totalling for 
          $195,233
        </h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Viewing ",
            undefined,
            " ",
            "beers",
            " totalling for ",
            "$195,233",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Viewing ",
          undefined,
          " ",
          "beers",
          " totalling for ",
          "$195,233",
        ],
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
